### PR TITLE
Fix entry selection in bootctl status

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/boot/bootctl.c [hunk 1]
```diff
@@ -682,12 +711,12 @@ static int boot_entry_show(
 }
 
 static int status_entries(
+                const BootConfig *config,
```
Comment (poettering) [inline src/boot/bootctl.c:714]: add an assert for this?

File: src/boot/bootctl.c [hunk 2]
```diff
@@ -709,21 +738,13 @@ static int status_entries(
                        SD_ID128_FORMAT_VAL(dollar_boot_partition_uuid));
         printf("\n\n");
 
-        r = boot_config_load(&config, esp_path, xbootldr_path);
-        if (r < 0)
-                return r;
-
-        r = boot_config_select_special_entries(&config);
-        if (r < 0)
-                return r;
```
Comment (poettering) [inline src/boot/bootctl.c:718]: hmm, why not call boot_config_load_and_select() from here?

(I mean, i am not opposed to moving this, I am just wondering, given that `status_entries()` only has a single caller)

Comment (keszybz) [reply src/boot/bootctl.c:718]: I wanted to keep the loading and printing separate. Looking at it again, either option should work about as well…

**Comment by keszybz** [comment]
centois (arch linux + sanitizers): ninja-test_sanitizers → 385/1297 test-sd-device   SIGABRT.
Doesn't look related, but it's suspious anyway.

I'll update with the added assert.

**Pull request merged**